### PR TITLE
Clean up standalone conversion tool

### PR DESCRIPTION
**Pull request by hurf**
Follow up #11368

Remove kube-version-change for all its functionalities are covered
by kubectl convert command. Also changed the related docs.

### CODE REVIEW

**Comment by hurf** [comment]
For converting config files from beta versions to v1, neither `kube-version-change` nor `kubectl convert` would help in current k8s version. So I just deleted the description about it in "api.md".

File: docs/admin/cluster-management.md [hunk 1]
```diff
@@ -212,13 +212,13 @@ for changes to this variable to take effect.
...
-You can use the `kube-version-change` utility to convert config files between different API versions.
+You can use `kubectl convert` command to convert config files between different API versions.
 
 ```console
-$ hack/build-go.sh cmd/kube-version-change
-$ _output/local/go/bin/kube-version-change -i myPod.v1beta3.yaml -o myPod.v1.yaml
+$ kubectl convert -f pod.yaml --output-version v1
 ```
 
+For more options, please refer to the usage of `kubectl convert` command.
```
Comment (bgrant0607) [inline docs/admin/cluster-management.md]: Please link to the documentation for convert.

**Comment by janetkuo** [comment]
`kube-version-change` tool is referenced in `docs/devel/adding-an-APIGroup.md`. Please update this document too. 

cc @caesarxuchao PTAL too.

**Comment by caesarxuchao** [comment]
I think we can safely remove the standalone tool. Thank you for taking care of it.

**Comment by janetkuo** [comment]
Need rebase now. Will apply tag afterwards.

**Comment by wojtek-t** [comment]
@hurf - can you please rebase?

**Comment by hurf** [comment]
@k8s-bot test this please

**Comment by wojtek-t** [comment]
LGTM

**Pull request merged**